components/feature/push/src/main/java/mozilla/components/feature/push/Connection.kt

```diff
@@ -106,26 +146,32 @@ internal class RustPushConnection(
     override suspend fun verifyConnection(): Boolean = synchronized(this) {
         val pushApi = api
         check(pushApi != null) { "Rust API is not initiated; updateToken hasn't been called yet." }
-        pushApi.verifyConnection()
+
+        return pushApi.verifyConnection()
     }
 
     @GuardedBy("this")
-    override fun decrypt(
+    override suspend fun decryptMessage(
         channelId: String,
         body: String,
         encoding: String,
         salt: String,
         cryptoKey: String
-    ): ByteArray = synchronized(this) {
+    ): Pair<PushScope, ByteArray> = synchronized(this) {
         val pushApi = api
         check(pushApi != null) { "Rust API is not initiated; updateToken hasn't been called yet." }
-        return pushApi.decrypt(channelID = channelId, body = body, encoding = encoding, salt = salt, dh = cryptoKey)
+
+        val scope = pushApi.dispatchInfoForChid(channelId)?.scope.orEmpty()
```

Comment:
So `scope` here could be empty? And the pair below would essentially by `("", 'some data')`?

Is this something you expect to see? seems like if this happens, we can't meaningfully process the decrypted message - how can consumers determine that the message is directed at them?

Reply:
> Is this something you expect to see? 

This can happen if there there is some corrupt subscription in the internal database from what I understand.

> seems like if this happens, we can't meaningfully process the decrypted message - how can consumers determine that the message is directed at them?

Messages can still be decrypted since the scope is only required for the final delivery and not for the decryption. A consumer would simply miss that message.

Reply:
"Missing a message" is partially what I meant by "we can't meaningfully process the decrypted message". What's the point of sending this message onto observers if we know they can't process it anyway?

Reply:
Ah fair, I guess we can use the lack of chid/scope to avoid decrypting the message and notifying clients since that probably means there are bigger problems if we get a valid message but the chid/scope doesn't exist.